cni/pkg/nodeagent/informers.go

```diff
@@ -258,54 +271,40 @@ func (s *InformerHandlers) reconcilePod(input any) error {
-			// then install Istio, then the pod goes ready, and we'd miss capture) - but that
-			// seems vanishingly unlikely
-			wasReady := kube.CheckPodReadyOrComplete(oldPod)
-			isReady := kube.CheckPodReadyOrComplete(newPod)
-			if wasReady != nil && isReady != nil && isAnnotated {
-				log.Infof("pod update event skipped: added/labeled by CNI plugin")
-				return nil
-			}
-
-			// netns == ""; at this point netns should have been added via the initial snapshot,
-			// or via the cni plugin. If it happens to get here before the cni plugin somehow,
-			// then we will just fail to add the pod to the mesh, and it will be retried later when cni plugin adds it.
+		}
 
-			// We need a pod IP - if the pod was added via the CNI plugin, that plugin told us the IPs
-			// for the pod. If this is a pod added via informer, the pod should have already gone thru
-			// the CNI plugin chain, and have a PodIP.
-			//
-			// If PodIPs exists, it is preferred, otherwise fallback to PodIP.
-			//
-			// If we get to this point and have a pod that really and truly has no IP in either of those,
-			// it's not routable at this point and something is wrong/we should discard this event.
-			podIPs := util.GetPodIPsIfPresent(pod)
-			if len(podIPs) == 0 {
-				log.Debugf("pod update event skipped: no IP assigned yet")
-				return nil
-			}
+		// netns == ""; at this point netns should have been added via the initial snapshot,
+		// or via the cni plugin. If it happens to get here before the cni plugin somehow,
+		// then we will just fail to add the pod to the mesh, and it will be retried later when cni plugin adds it.
+
+		// We need a pod IP - if the pod was added via the CNI plugin, that plugin told us the IPs
+		// for the pod. If this is a pod added via informer, the pod should have already gone thru
+		// the CNI plugin chain, and have a PodIP.
+		//
+		// If PodIPs exists, it is preferred, otherwise fallback to PodIP.
+		//
+		// If we get to this point and have a pod that really and truly has no IP in either of those,
+		// it's not routable at this point and something is wrong/we should discard this event.
+		podIPs := util.GetPodIPsIfPresent(newPod)
+		if len(podIPs) == 0 {
+			log.Debugf("pod update event skipped: no IP assigned yet")
+			return nil
+		}
 
-			log.Debugf("pod is now enrolled, adding to mesh")
-			err := s.dataplane.AddPodToMesh(s.ctx, pod, podIPs, "")
-			if err != nil {
-				log.Warnf("AddPodToMesh returned: %v", err)
-			}
+		log.Debugf("pod is now enrolled, adding to mesh")
+		err := s.dataplane.AddPodToMesh(s.ctx, newPod, podIPs, "")
+		if err != nil {
+			log.Warnf("AddPodToMesh returned: %v", err)
 		}
 	case controllers.EventDelete:
-		// We are the only thing that should be annotating the pods for mesh inclusion.
-		// If we did, remove it from ztunnel
-		if util.PodRedirectionActive(pod) {
+		// If the pod was annotated (by informer or plugin) remove pod from mesh.
+		if util.PodRedirectionActive(newPod) {
```

Comment:
This will always return `false` since newPod will always be nil since its a delete and we try to Get() it

Reply:
Yeah - overly aggressive with changing `delete` here, ty for catching.